175 Welcome Screen Styling

diegolucasb opened the pull request:
Fixes #175 

I'd like to make a few observations:

- For the obvious reasons, I've tried to use vector for nessie and name logo (Firefox lockbox), but the XML vector file generated by Android Studio was getting this lint error:  
```Very long vector path (1115 characters), which is bad for performance. Considering reducing precision, removing minor details or rasterizing vector.```
So, as our vector image seems to complex, I decided to use PNG files. According to [this post](https://medium.com/upday-devs/optimizing-the-performance-of-vector-drawables-680a4c456286), in those cases, it is better to use a raster image. Please, it would be great if your design team could take a look at `nessie.svg` and `logo.svg` files in order to check if they can reduce vector path precision, otherwise, I recommend we try using [WebP images](https://developers.google.com/speed/webp/)

- On Zeplin, the status bar is gray. I was not sure if it was something to change, so I didn't
- I tried to follow the patterns I could realize you guys were using, like snake case for resources and camel case for layout’s ids.
- I’ve seen that `fragment_locked` layout need to be changed as well, but as it isn’t part of this issue, I didn’t want to change it. As soon as this PR gets merged, I’d like to work on this lock screen, if you don’t mind!


**Please, let me know if it is all according to your expectations and if there is anything that needs to be changed**

## Screenshots or Videos
![photo_2018-11-20_21-00-32](https://user-images.githubusercontent.com/4621339/48808239-574cb900-ed07-11e8-829b-f741816b5f51.jpg)

## To Do

- [x] add “WIP” to the PR title if pushing up but not complete nor ready for review
- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] ~add unit tests~
- [x] consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- [x] request the "UX" team perform a design review (if/when applicable)
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)
- [x] check on the [accessibility](https://mozilla-lockbox.github.io/lockbox-android/accessibility/) of any added UI

eliserichards commented:
Hi Diego! I saw your comment about the "very long vector path" failure. These kinds of errors usually live in the `lint-baseline` file. To fix this, delete your local file, then run `./gradlew lint` to check that the error was added to the baseline file correctly.

diegolucasb commented:
Thanks, @eliserichards! I will fix it and commit the changes!

diegolucasb commented:
Guys please, help me understand what I should do about the images.  Do I use AndroidStudio generated the SVG file and add its issue on `lint-baseline.xml`? 

And what about Bitrise, what went wrong? If I run `./gradlew check` on this branch, it builds successfully.

Thank.

jhugman commented on .idea/copyright/Mozilla.xml:
```diff
@@ -1,6 +0,0 @@
-    <option name="myName" value="Mozilla" />
-    <option name="notice" value="This Source Code Form is subject to the terms of the Mozilla Public&#10;License, v. 2.0. If a copy of the MPL was not distributed with this&#10;file, You can obtain one at http://mozilla.org/MPL/2.0/." />
-  </copyright>
-</component>
```
We rather like these `.idea` files. Could you unstage them from your PR?

diegolucasb replied:
Removed

devinreams commented:
Hey @diegolucasb thank you very much for your contribution!

You made the right decision on the vector assets. We just ran into this ourselves recently and decided to document our image/asset practices here: 
https://github.com/mozilla-lockbox/lockbox-android/blob/master/docs/add_image_assets.md

In short, Nessie is an OK use of a flat image in this case. Thank you for that. 👍 

As to the failing build (lint error) in bitrise, it appears to me that the use of `lineheight` is the only issue. We’re aiming to support API 23+.

```
* What went wrong:
Execution failed for task ':app:lintDebug'.
> Lint found errors in the project; aborting build.
  
  Fix the issues identified by lint, or add the following to your build script to proceed with errors:
  ...
  android {
      lintOptions {
          abortOnError false
      }
  }
  ...
  
  Errors found:
  
  /bitrise/src/app/src/main/res/layout/fragment_welcome.xml:54: Error: Attribute lineHeight is only used in API level 28 and higher (current min is 23) [UnusedAttribute]
          android:lineHeight="19.5sp"
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /bitrise/src/app/src/main/res/layout/fragment_welcome.xml:73: Error: Attribute lineHeight is only used in API level 28 and higher (current min is 23) [UnusedAttribute]
          android:lineHeight="19.5sp"
          ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Please let us know if you’d like any of our help on this branch itself, otherwise we’ll assume you’re able and willing to make the few remaining changes here. Thank you again for your contribution! 🏅

diegolucasb commented:
@jhugman sorry for that! I didn't realise these `.idea` files were committed;

@devinreams, In this case, I think we have two options, either add linter exception or add `tools:targetApi="M"` Well, I don't think it noth worth, so I removed the attribute, as I couldn't see any visual change. If you think there other better way please, let'me know and I make the changes!

devinreams commented:
> I removed the attribute, as I couldn't see any visual change

Ah okay, that seems reasonable. Thank you!

I've re-requested a quick code review and approval from the mobile engineering team. I also just requested the UX team to test this out and make sure it's looking A-OK. 🎨

diegolucasb commented:
Thank you @nickbrandt! I'll fix it!

diegolucasb commented:
@nickbrandt 

> On very small devices using the simulator, the Nessie illustration with waves doesn't quite line up (not sure how common this small device size is)

I created `dimens.xml` file for small devices and I made include_nessie use margins values from it

> The copy under the button ("To use Firefox Lockbox...") appears to be Medium or Semi-bold instead of Regular

It was medium indeed. I made it regular

> The press state for the button looks like it adds a shadow. Can we update it to reflect what we have in this pattern library (apologies for not referencing this for you earlier): https://zpl.io/bPmrX6x
Essentially we are looking for the press state to animate to being the very light blue color.

No problem. I removed the shadow (when the button was pressed) and changed its selector to be light blue (#cce5f1) when pressed.

As I can't see Bitrise log, please let me know what went wrong if it fails.
Thank you!

sashei commented:
@diegolucasb It looks like bitrise is failing because of the merge conflicts right now.

sashei commented:
@diegolucasb next failure: linter! :)

nickbrandt commented:
@diegolucasb looks good, thanks again for your help on this!
![screenshot_1543423416](https://user-images.githubusercontent.com/5185090/49167535-ca02f900-f2f2-11e8-9208-1df97ed9f07d.png)

diegolucasb commented:
> @diegolucasb next failure: linter! :)

@sashei  would you mind telling me what had failed?
When I run `./gradlew check` it builds successfully for me.

sashei commented:
```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:lintDebug'.
> Lint found errors in the project; aborting build.
  
  Fix the issues identified by lint, or add the following to your build script to proceed with errors:
  ...
  android {
      lintOptions {
          abortOnError false
      }
  }
  ...
  
  Errors found:
  
  /bitrise/src/app/src/main/res/values-ldpi/dimens.xml:8: Error: The resource R.dimen.nav_padding_lock_now appears to be unused [UnusedResources]
      <dimen name="nav_padding_lock_now" tools:override="true">125dp</dimen>
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~
  
```

@diegolucasb ^

devinreams commented:
Thank you again for your contributions here @diegolucasb! 🙏 

It looks great and we'll be sure to get you on the list to help test an early version soon...
